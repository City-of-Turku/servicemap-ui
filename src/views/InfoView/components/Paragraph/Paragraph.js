import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

const Paragraph = ({
  classes, intl, isTitle, translationId,
}) => (
  <div className={isTitle ? classes.title : classes.text}>
    <Typography component={isTitle ? 'h3' : 'p'} variant="body2" aria-label={intl.formatMessage({ id: translationId })}>
      {intl.formatMessage({ id: translationId })}
    </Typography>
  </div>
);

Paragraph.propTypes = {
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  isTitle: PropTypes.bool,
  translationId: PropTypes.string,
};

Paragraph.defaultProps = {
  isTitle: false,
  translationId: '',
};

export default Paragraph;
