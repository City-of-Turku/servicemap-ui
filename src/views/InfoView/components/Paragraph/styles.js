export default theme => ({
  title: {
    paddingBottom: theme.spacing(0.1),
  },
  text: {
    padding: theme.spacing(2),
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  },
});
