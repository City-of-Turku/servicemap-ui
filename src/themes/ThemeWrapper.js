import { ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';
import themes from '.';
import config from '../../config';
import { selectThemeMode } from '../redux/selectors/user';

// If external theme (by Turku) is true, then can be used to select which theme to get
const externalTheme = config.themePKG;
const isExternalTheme = !externalTheme || externalTheme === 'undefined' ? null : externalTheme;

/**
 * Returns default theme based on env value
 * @returns theme
 */
const getTheme = () => {
  if (isExternalTheme) {
    return themes.SMThemeTku;
  }
  return themes.SMTheme;
};

// Component to handle theme changes
const ThemeHandler = ({ children }) => {
  const themeMode = useSelector(selectThemeMode);
  return (// Get correct theme setting from store
    <ThemeProvider theme={themeMode === 'dark' ? themes.SMThemeDark : getTheme()}>
      {children}
    </ThemeProvider>
  );
};
ThemeHandler.propTypes = {
  children: PropTypes.node,
};
ThemeHandler.defaultProps = {
  children: null,
};

export default ThemeHandler;
