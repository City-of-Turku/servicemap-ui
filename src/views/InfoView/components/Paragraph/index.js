import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import Paragraph from './Paragraph';
import styles from './styles';

export default injectIntl(withStyles(styles)(Paragraph));
