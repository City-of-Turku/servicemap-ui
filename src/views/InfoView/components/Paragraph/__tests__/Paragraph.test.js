// Link.react.test.js
import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../themes';
import Paragraph from '../index';
import finnishTranslations from '../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  isTitle: false,
  translationId: 'info.view.serviceInfo',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>
      {children}
    </MuiThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<Paragraph />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<Paragraph {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<Paragraph {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(finnishTranslations['info.view.serviceInfo']);
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<Paragraph {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toContain(finnishTranslations['info.view.serviceInfo']);
  });
});
