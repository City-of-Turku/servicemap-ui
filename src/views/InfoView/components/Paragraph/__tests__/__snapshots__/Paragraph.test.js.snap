// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should work 1`] = `
<div>
  <div
    class="Paragraph-text-2"
  >
    <p
      aria-label="Palvelukartalta löytyvät Turun julkiset toimipisteet ja niiden palvelut, esimerkiksi koulut, päiväkodit ja terveysasemat."
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Palvelukartalta löytyvät Turun julkiset toimipisteet ja niiden palvelut, esimerkiksi koulut, päiväkodit ja terveysasemat.
    </p>
  </div>
</div>
`;
